### PR TITLE
Point Fix GetRouteClient Error Checking

### PR DESCRIPTION
**Pull request by mphammer**
Check for more error messages before returning a RouteClient

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/342#" title="Author self-approved">mphammer</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.4.x/OWNERS)~~ [mphammer]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/util/common.go [hunk 1]
```diff
@@ -988,8 +988,8 @@ func GetRouteClient(restConfig *rest.Config) *routeclient.RouteV1Client {
 		log.Errorf("unable to get route client")
 		return nil
 	}
-	_, err = GetRoute(routeClient, "default", "docker-registry")
-	if err != nil && strings.Contains(err.Error(), "could not find the requested resource") && strings.Contains(err.Error(), "openshift.io") {
+	_, err = ListRoutes(routeClient, "default", "")
+	if err != nil && (strings.Contains(err.Error(), "could not find the requested resource") || strings.Contains(err.Error(), "cannot list routes")) && strings.Contains(err.Error(), "openshift.io") {
```
Comment (rrati) [inline pkg/util/common.go]: Let's not look for fixed strings.  That's very fragile.  If there's an error getting routes then the route client is disabled.  Is there a need for more complexity?

Comment (msenmurugan) [reply pkg/util/common.go]: For get, we need this complexity bcoz of openshift versions.. But for list, you are right, we don't need to compare

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/342#" title="Author self-approved">mphammer</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/342#pullrequestreview-237278507" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/release-2019.4.x/OWNERS)~~ [mphammer,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: new pull request created: #344

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/342#issuecomment-492258807):

>/cherrypick master


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**